src/identifiers.md

```diff
@@ -60,4 +63,45 @@ See the [HIR chapter][hir-map] for more detailed information.
 
 ## In the MIR
 
-**TODO**
+- [`BasicBlock`] identifies a *basic block*. It points to an instance of
+  [`BasicBlockData`], which can be retrieved by indexing into
+  [`Body::basic_blocks()`] (note that you must call a function; the field is
+  private).
+
+- [`Local`] identifies a local variable in a function. Its associated data is in
+  [`LocalDecl`], which can be retrieved by indexing into [`Body.local_decls`].
+
+- [`Field`] identifies a struct's, union's, or enum variant's field. It is used
+  as a "projection" in [`Place`].
+
+- [`SourceScope`] identifies a name scope in the original source code. Used for
+  diagnostics and for debuginfo in debuggers. It points to an instance of
+  [`SourceScopeData`], which can be retrieved by indexing into
+  [`Body.source_scopes`].
+
+- [`Promoted`] identifies a promoted constant within another item (related to
+  const evaluation). Note: it is unique only locally within the item, so it should be associated with a `DefId`;
+  [`GlobalId`] must be used if you need to *globally* identify the promoted.
```

Comment:
@oli-obk The change you applied confuses me: the text now says

> Note: it is unique only locally within the item, so it should be associated with a `DefId`;
> [`GlobalId`] must be used if you need to *globally* identify the promoted.

So which one is it? It says that you should associate it with a `DefId`, but it also says use `GlobalId`.

Reply:
`GlobalId` is the combination of `DefId` + `Substs` + `Option<Promoted>`. You can identify a promoted with just a `DefId` and `Promoted`, but you may not be able to compute its value without the `Substs`, because it's still polymorphic.

Reply:
So what should I change this sentence to?

Reply:
I guess something like `must be used if you need a globally unique identifier for the monomorphic version of a promoted`? Is that filled with too many weird words or do you think someone who gets to this section the first time will be able to understand it?

Reply:
Hmm, but didn't you say that you can identify a promoted with just `DefId` and `Promoted`, and you only need `GlobalId` in order to actually compute the value?

Reply:
My mental model was always that `GlobalId` identifies the monomorphic version of sth, `DefId` identifies a definition (potentially polymorphic) and `(DefId, Promoted)` identifies a promoted within a definition (also potentially polymorphic). But "identifying a monomorphic version" may not really be a thing...

Reply:
You know what, how about we just say "`GlobalId` gives you more information (TODO)" and come back to this later? That way we can get the rest of it merged :)

Reply:
I personally wouldn't mind, especially given that this PR itself removes a giant **TODO** :)